From: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Date: Wed, 22 Jan 2020 12:39:49 +0100
Subject: [PATCH] Add dynamic symbols linking with cmake too

Upstream-Status: Submitted [https://github.com/eclipse/mosquitto/pull/1571]
Signed-off-by: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Last-Update: 2023-07-09
-

Signed-off-by: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
---
 lib/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 31cc35e..f0f7113 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -94,6 +94,8 @@ set_target_properties(libmosquitto PROPERTIES
 	OUTPUT_NAME mosquitto
 	VERSION ${VERSION}
 	SOVERSION 1
+	LINK_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/linker.version
+	LINK_FLAGS "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/linker.version"
 )
 
 install(TARGETS libmosquitto
