From 66c62767354f986cad928779027eb7d5182c77c8 Mon Sep 17 00:00:00 2001
From: "Roger A. Light" <roger@atchoo.org>
Date: Tue, 12 Sep 2023 23:11:54 +0100
Subject: [PATCH] Fix protocol

---
 docker/1.5-openssl/Dockerfile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/docker/1.5-openssl/Dockerfile b/docker/1.5-openssl/Dockerfile
index 69ade5408..afbaa7458 100644
--- a/docker/1.5-openssl/Dockerfile
+++ b/docker/1.5-openssl/Dockerfile
@@ -45,7 +45,7 @@ RUN set -x && \
     export GNUPGHOME="$(mktemp -d)" && \
     found=''; \
     for server in \
-        htps://keys.openpgp.org \
+        hkps://keys.openpgp.org \
         hkp://keyserver.ubuntu.com:80 \
         pgp.mit.edu \
     ; do \
