From: Gianfranco Costamagna <locutusofborg@debian.org>
Description: Use system pico library
--- mosquitto-2.1.2.orig/lib/CMakeLists.txt
+++ mosquitto-2.1.2/lib/CMakeLists.txt
@@ -74,11 +74,19 @@ endif()
 
 if(WITH_WEBSOCKETS AND WITH_WEBSOCKETS_BUILTIN)
 	add_definitions("-DWITH_WEBSOCKETS=WS_IS_BUILTIN")
-	set(C_SRC ${C_SRC}
-		../deps/picohttpparser/picohttpparser.c
-		http_client.c http_client.h
-		net_ws.c
-		)
+	find_library(PICOHTTPPARSER_LIBRARY NAMES picohttpparser)
+	if(PICOHTTPPARSER_LIBRARY)
+		set(C_SRC ${C_SRC}
+			http_client.c http_client.h
+			net_ws.c
+			)
+	else()
+		set(C_SRC ${C_SRC}
+			../deps/picohttpparser/picohttpparser.c
+			http_client.c http_client.h
+			net_ws.c
+			)
+	endif()
 endif()
 
 add_library(libmosquitto SHARED
@@ -86,8 +94,12 @@ add_library(libmosquitto SHARED
 )
 
 if(WITH_WEBSOCKETS AND WITH_WEBSOCKETS_BUILTIN)
-	target_include_directories(libmosquitto PRIVATE
-		"${mosquitto_SOURCE_DIR}/deps/picohttpparser")
+	if(NOT PICOHTTPPARSER_LIBRARY)
+		target_include_directories(libmosquitto PRIVATE
+			"${mosquitto_SOURCE_DIR}/deps/picohttpparser")
+	else()
+		target_link_libraries(libmosquitto PRIVATE ${PICOHTTPPARSER_LIBRARY})
+	endif()
 endif()
 
 target_include_directories(libmosquitto
--- mosquitto-2.1.2.orig/src/CMakeLists.txt
+++ mosquitto-2.1.2/src/CMakeLists.txt
@@ -231,8 +231,15 @@ endif()
 
 if(WITH_WEBSOCKETS)
 	if(WITH_WEBSOCKETS_BUILTIN)
+	find_library(PICOHTTPPARSER_LIBRARY NAMES picohttpparser)
+		if(PICOHTTPPARSER_LIBRARY)
+			target_link_libraries(mosquitto PRIVATE ${PICOHTTPPARSER_LIBRARY})
+		else()
+			target_sources(mosquitto PRIVATE ${mosquitto_SOURCE_DIR}/deps/picohttpparser/picohttpparser.c)
+			target_include_directories(mosquitto PRIVATE
+				"${mosquitto_SOURCE_DIR}/deps/picohttpparser")
+		endif()
 		target_compile_definitions(mosquitto PRIVATE "WITH_WEBSOCKETS=WS_IS_BUILTIN")
-		target_sources(mosquitto PRIVATE ${mosquitto_SOURCE_DIR}/deps/picohttpparser/picohttpparser.c)
 	else()
 		find_package(libwebsockets)
 		target_compile_definitions(mosquitto PRIVATE "WITH_WEBSOCKETS=WS_IS_LWS")
@@ -244,10 +251,7 @@ if (ANDROID)
 endif (ANDROID)
 
 if(WITH_WEBSOCKETS)
-	if(WITH_WEBSOCKETS_BUILTIN)
-		target_include_directories(mosquitto PRIVATE
-			"${mosquitto_SOURCE_DIR}/deps/picohttpparser")
-	else()
+	if(NOT WITH_WEBSOCKETS_BUILTIN)
 		if(STATIC_WEBSOCKETS)
 			target_link_libraries(mosquitto PRIVATE websockets)
 			if(WIN32)
